File: ase/misc/sort.js

var Comparable = GCJ["java.lang.Comparable"];
var Arrays = GCJ["java.util.Arrays"];

function Foo(value) {
  this.value = value;
  this.compareTo = function(x) {
    var xv = x.value;
    print("compareTo " + value + " " + xv);
    if (value < xv) { return -1; }
    if (value > xv) { return 1; }
    return 0;
  };
}

var len = 20;
var arr = Comparable.NewArray(len);
for (var i = 0; i < len; ++i) {
  arr[i] = new Foo(Math.round(Math.random() * 1000));
  print("i=" + i + " v=" + arr[i].value);
}

Arrays.sort(arr);

for (var i = 0; i < len; ++i) {
  print("i=" + i + " v=" + arr[i].value);
}

File: ase/misc/java/sort.lua

local Lang = Java.java.lang;
local Comparable = Lang.Comparable;
local Util = Java.java.util;
local Arrays = Util.Arrays;

function Foo(value)
  local o = {}
  o.value = value
  o.compareTo = function(x)
    local xv = x.value
    print("compareTo: " .. value .. " " .. xv)
    if (value < xv) then return -1 end
    if (value > xv) then return 1 end
    return 0
  end
  return o
end

math.randomseed(os.time())

local len = 20
local arr = Comparable.NewArray(len)
local i
for i = 0, len - 1 do
  arr[i] = Comparable(Foo(math.ceil(math.random() * 1000)))
  print("i=" .. i .. " v=" .. arr[i].value)
end

Arrays.sort(arr)

for i = 0, len - 1 do
  print("i=" .. i .. " v=" .. arr[i].value)
end

